src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -692,3 +588,121 @@ def update_connectedk8s(cmd, instance, tags=None):
+    except Exception as e:
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception=e, fault_type=consts.Load_Kubeconfig_Fault_Type,
+                                summary='Problem loading the kubeconfig file')
+        raise CLIError("Problem loading the kubeconfig file." + str(e))
+    configuration = kube_client.Configuration()
+
+    # Checking the connection to kubernetes cluster.
+    # This check was added to avoid large timeouts when connecting to AAD Enabled AKS clusters
+    # if the user had not logged in.
+    check_kube_connection(configuration)
+
+    # Get kubernetes cluster info for telemetry
+    kubernetes_version = get_server_version(configuration)
+    kubernetes_distro = get_kubernetes_distro(configuration)
+
+    kubernetes_properties = {
+        'Context.Default.AzureCLI.KubernetesVersion': kubernetes_version,
+        'Context.Default.AzureCLI.KubernetesDistro': kubernetes_distro
+    }
+    telemetry.add_extension_event('connectedk8s', kubernetes_properties)
+
+    # Checking helm installation
+    check_helm_install(kube_config, kube_context)
+
+    # Check helm version
+    helm_version = check_helm_version(kube_config, kube_context)
+    telemetry.add_extension_event('connectedk8s', {'Context.Default.AzureCLI.HelmVersion': helm_version})
+
+    # Validate location
+    utils.validate_location(cmd, location)
+
+    # Check whether Connected Cluster is present
+    if not connected_cluster_exists(client, resource_group_name, cluster_name):
+        telemetry.set_user_fault()
+        telemetry.set_exception(exception='The connected cluster resource does not exist', fault_type=consts.Resource_Does_Not_Exist_Fault_Type,
+                                summary='Connected cluster resource does not exist')
+        raise CLIError("The connected cluster resource {} does not exist ".format(cluster_name) +
+                       "in the resource group {} ".format(resource_group_name) +
+                       "Please onboard the connected cluster using az connectedk8s command")
+
+    # Fetch Connected Cluster for agent version
+    connected_cluster = get_connectedk8s(cmd, client, resource_group_name, cluster_name)
+
+    # Adding helm repo
+    utils.add_helm_repo(kube_config, kube_context)
+
+    # Retrieving Helm chart OCI Artifact location
+    registry_path = os.getenv('HELMREGISTRY') if os.getenv('HELMREGISTRY') else utils.get_helm_registry(profile, location)
+
+    # Set agent version in registry path
+    if connected_cluster.agent_version is not None:
+        if not utils.is_update_allowed(connected_cluster.agent_version):
+            telemetry.set_user_fault()
+            message = str.format(consts.Update_Not_Allowed, "0.2.5", "0.1.214-dev")
+            telemetry.set_exception(exception=message, fault_type=consts.Update_Not_Allowed_Fault_Type,
+                                    summary='Update not allowed')
+            raise CLIError(message)
```

Comment:
Check comment above on is_update_allowed. Let's discuss tomorrow if we need this